controllers/rcj_soccer_referee_supervisor/referee/referee.py

```diff
@@ -1,44 +1,248 @@
 import random
-from typing import Optional
+import struct
+from typing import List, Optional, Tuple
+
+from controller import Supervisor
 
 from referee.consts import (
+    BALL_INITIAL_TRANSLATION,
     GameEvents,
+    KICKOFF_TRANSLATION,
     LACK_OF_PROGRESS_NUMBER_OF_NEUTRAL_SPOTS,
+    MAX_EVENT_MESSAGES_IN_QUEUE,
     NeutralSpotDistanceType,
+    ROBOT_INITIAL_ROTATION,
+    ROBOT_INITIAL_TRANSLATION,
     ROBOT_NAMES,
     Team,
     TIME_STEP,
 )
-from referee.supervisor import RCJSoccerSupervisor
-from referee.utils import is_in_blue_goal, is_in_yellow_goal, is_outside
+from referee.event_handlers import EventHandler
+from referee.eventer import Eventer
+from referee.penalty_area_checker import PenaltyAreaChecker
+from referee.progress_checker import ProgressChecker
+from referee.utils import (
+    is_in_blue_goal,
+    is_in_yellow_goal,
+    is_outside,
+    time_to_string,
+)
+
+
+class RCJSoccerReferee:
+    def __init__(
+        self,
+        supervisor: Supervisor,
+        match_time: int,
+        match_id: int,
+        half_id: int,
+        progress_check_steps: int,
+        progress_check_threshold: int,
+        ball_progress_check_steps: int,
+        ball_progress_check_threshold: int,
+        team_name_blue: str,
+        team_name_yellow: str,
+        initial_score_blue: int,
+        initial_score_yellow: int,
+        penalty_area_allowed_time: int,
+        penalty_area_reset_after: int,
+        post_goal_wait_time: int = 3,
+        initial_position_noise: float = 0.15,
+    ):
+        self.sv = supervisor
+        self.match_time = match_time
+        self.time = match_time
+        self.match_id = match_id
+        self.half_id = half_id
+        self.team_name_blue = team_name_blue
+        self.team_name_yellow = team_name_yellow
+        self.score_blue = initial_score_blue
+        self.score_yellow = initial_score_yellow
+        self.post_goal_wait_time = post_goal_wait_time
+        self.initial_position_noise = initial_position_noise
+
+        self.ball_reset_timer = 0
+        self.ball_stop = 2
+
+        self.robot_in_penalty_counter = {}
+        self.progress_check = {}
+        self.penalty_area_check = {}
+        for robot in ROBOT_NAMES:
+            self.progress_check[robot] = ProgressChecker(
+                progress_check_steps, progress_check_threshold
+            )
+
+            self.penalty_area_check[robot] = PenaltyAreaChecker(
+                penalty_area_allowed_time,
+                penalty_area_reset_after,
+            )
+
+            self.robot_in_penalty_counter[robot] = 0
+
+        self.progress_check["ball"] = ProgressChecker(
+            ball_progress_check_steps, ball_progress_check_threshold
+        )
+
+        self.eventer = Eventer()
+        # Event message queue to be drawn
```

Comment:
```suggestion
        # Event message queue to be drawn from
```

I honestly do not know -- just wanted to flag this (and realize it's been in there elsewhere already)

Reply:
Done